metacat-common-server/src/main/java/com/netflix/metacat/common/server/usermetadata/BusinessMetadataRules.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.metacat.common.server.usermetadata;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import javax.annotation.Nullable;
+
+/**
+ * Business Metadata Rules.
+ *
+ * @author zhenl
+ * @since 1.2.0
+ */
+public interface BusinessMetadataRules {
+    /**
+     * apply business rules before retrieving back. These rules may change or overriding existing
+     * business metadata.
+     *
+     * @param objectNode obj node
+     * @return result object node
+     */
+    default ObjectNode applyRules(final ObjectNode objectNode) {
+        return objectNode;
+    }
+
+    /**
+     * apply business rules before retrieving back. These rules may change or overriding existing
+     * business metadata.
+     *
+     * @param objectNode obj node
+     * @param guideNode  this will be an input node for the rules
+     * @return result object node
+     */
+    default ObjectNode applyRules(final ObjectNode objectNode,
+                                  @Nullable final ObjectNode guideNode) {
+        return objectNode;
+    }
+
+    /**
+     * valide ObjectNode before storing it.
+     *
+     * @param objectNode obj node
+     * @return Boolean
+     */
+    default boolean validateBusinessMetadata(final ObjectNode objectNode) {
```

Comment:
I thought we will apply rules and if validated will store it. Do we need the above two methods?